src/vulkan/graphics_pipeline.cc

```diff
@@ -930,18 +930,15 @@ Result GraphicsPipeline::Draw(const DrawArraysCommand* command,
 
 Result GraphicsPipeline::ProcessCommands() {
   DeactivateRenderPassIfNeeded();
-
   return Pipeline::ProcessCommands();
 }
 
 void GraphicsPipeline::Shutdown() {
   DeactivateRenderPassIfNeeded();
 
-  if (index_buffer_)
-    index_buffer_->Shutdown();
-
+  index_buffer_ = nullptr;
   Pipeline::Shutdown();
-  frame_->Shutdown();
```

Comment:
c++ learner question: here you transform `frame_->Shutdown()` to `frame_ = nullptr`, does this force the call fo frame_ destructor?

Reply:
Yea, frame_ is a std::unique_ptr<FrameBuffer> by setting the unique_ptr to null it will destroy the FrameBuffer object that it currently has and store the nullptr.

Note, this wouldn't work if frame_ wasn't a unique_ptr.